case_studies/adaptive_tiling/region_encoder.py

```diff
@@ -79,52 +79,18 @@ def convert_locs_padded_to_unpadded(self, locs):
 
         return locs
 
-    def variational_mode(
-        self, pred: Dict[str, Tensor], return_full: bool | None = True
-    ) -> FullCatalog | TileCatalog:
-        """Compute the mode of the variational distribution.
-
-        Args:
-            pred (Dict[str, Tensor]): model predictions
-            return_full (bool, optional): Returns a FullCatalog if true, otherwise returns a
-                RegionCatalog. Defaults to True.
-
-        Returns:
-            Union[FullCatalog, TileCatalog]: Catalog based on predictions.
-        """
-        # Get point estimate of each distribution
-        tile_is_on_array = pred["on_prob"].mode
-        locs = pred["loc"].mode
-        aux_var = pred["aux_var"].probs[..., 1]
-
-        star_fluxes = torch.stack(
-            [pred[name].mode * tile_is_on_array for name in self.STAR_FLUX_NAMES], dim=3
-        )
-        galaxy_fluxes = torch.stack(
-            [pred[name].mode * tile_is_on_array for name in self.GAL_FLUX_NAMES], dim=3
-        )
-
-        galaxy_bools = pred["galaxy_prob"].mode
-        source_type = SourceType.STAR * (1 - galaxy_bools) + SourceType.GALAXY * galaxy_bools
-
-        galsim_dists = [pred[f"galsim_{name}"] for name in self.GALSIM_NAMES]
-        galsim_param_lst = []
-        for d in galsim_dists:
-            # use median for transformed distributions since mode isn't implemented
-            est = d.icdf(torch.tensor(0.5)) if isinstance(d, TransformedDistribution) else d.mode
-            galsim_param_lst.append(est)
-        galaxy_params = torch.stack(galsim_param_lst, dim=3)
-
+    def suppress_tiles(self, d):
         # suppress tiles at boundaries/corners based on priority vars
-        batch_size, nth, ntw = tile_is_on_array.shape
+        batch_size, nth, ntw = d["tile_is_on_array"].shape
         threshold = (
             self.overlap_slen / 2 / self.tile_slen,
             1 - (self.overlap_slen / 2 / self.tile_slen),
         )
 
+        locs = d["locs"]
         n_rows, n_cols = nth * 2 - 1, ntw * 2 - 1
         for b, i, j in itertools.product(range(batch_size), range(nth), range(ntw)):
```

Comment:
This loop looks really slow. Is that the case? Can it not be done without a loop, using tensor operations?

Reply:
There probably is some clever way to check all cases (vertical boundary, horizontal boundary, corner) in a vectorized way, but I was having trouble getting it to work so I just used the naive loop version.

Functionally, it's not that bad - it takes 0.02 seconds on average for a 1x40x40 (tiles) image, so should generally be negligible.